language/compiler/ir_to_bytecode/src/context.rs

```diff
@@ -289,6 +299,33 @@ impl<'a> Context<'a> {
         Ok(())
     }
 
+    /// Add the source location to the bytecode instruction sequence starting at `offset` within
+    /// the function body given by the `FunctionDefinitionIndex`
+    pub fn add_source_info(
+        &mut self,
+        fdef_idx: FunctionDefinitionIndex,
+        offset: CodeOffset,
+        location: Loc,
+    ) {
+        self.source_map.add_code_mapping(fdef_idx, offset, location);
+    }
+
+    pub fn add_local_mapping(&mut self, fdef_idx: FunctionDefinitionIndex, variable: Var_) {
+        self.source_map.add_local_mapping(fdef_idx, variable)
+    }
+
+    pub fn incr_function_index(&mut self) {
+        match self.current_function_index {
+            FunctionDefinitionIndex(i) => {
+                self.current_function_index = FunctionDefinitionIndex(i + 1)
+            }
+        }
+    }
+
+    pub fn current_function_index(&self) -> FunctionDefinitionIndex {
```

Comment:
Is this unused? I don't see any code that calls it. What would happen if we deleted `current_function_index`?

Reply:
Agh! That was on the chopping block, and I meant to remove it. (Like I said, cleanup is needed....). If you remove it, nothing will break right now.